app/app.rb

```diff
@@ -112,8 +112,4 @@ def list_all_rentals
     end
     @parent.show_menu
```

Comment:
**NOTE:** I would have loved to highligth the actual portion but they're not highlightable because no changes were made to them. But this below illustration, you would get where exactly I am asking for the changes. Kindly look into all the suggestions and apply them into your code. It's all about your `app.rb`. Make it more dry by applying the example below to your code.

- [ ] Great job refactoring your code for good practice. However, there's little stuff to still look into.
- [ ] Your app.rb file still looks bukky, remember, all requests for input and input validation should've been placed in the main.rb file and/or classes/methods whose responsibilities are to request or validate input. I would suggest you create a separate file say `list.rb`. You can have all your methods in that file like below.

```
   class List
       def books(list)
          puts 'No books' if list.empty?
          list.each do |book|
             puts "Title: \"#{book.title}\", Author: #{book.author}"
         end
         puts 'Press Enter to continue!'
       end
```
Then, inside your `app.rb`, you can require the List class, then the methods inside the it to make your app to follow YAGNI, KISS, SOLID etc. Take a look for your `app.rb` :point_down: 

```
require './list'

class App
  attr_reader :options
.....
  def initialize(books_list = [], people_list = [])
    @books_list = books_list
    @people_list = people_list
    @options = {
      '1' => :list_books,
      '2' => :list_people,
      '3' => :create_person,
      '4' => :create_book,
      '5' => :create_rental,
      '6' => :list_rentals
      }
     @list = List.new
  end

    def list_books
       @list.books(@books_list)
        gets.chomp
    end

```
This would make your app to look more dry.